### ISSUE TITLE
Softmax attention-optimized kernel: hardcoded CB length 80 incompatible with block_size=3 or 6 in large-kernel fallback

### ISSUE THREAD

**Issue by djstefanovicTT** [tenstorrent/tt-metal#38626]
The attention-optimized softmax kernel crashes with TT_FATAL when the large-kernel fallback path is triggered and find_max_divisor(Wt, N) returns 3 or 6, because the hardcoded circular buffer length of 80 is not divisible by those values. Exact error is ` Always | TT_FATAL: Size of cb: 80 must be divisible by the size of block used by the reader and compute kernel. (assert.hpp:104)`

Python script that reproduces the problem:
```
import torch
import ttnn

device = ttnn.open_device(device_id=0)

x = torch.randn(1, 1, 32, 6240, dtype=torch.bfloat16)
x_tt = ttnn.from_torch(x, device=device, layout=ttnn.TILE_LAYOUT)

compute_config = ttnn.WormholeComputeKernelConfig(
    math_fidelity=ttnn.MathFidelity.HiFi4,
    fp32_dest_acc_en=True,
)
result = ttnn.softmax(
    x_tt,
    dim=-1,
    compute_kernel_config=compute_config,
    numeric_stable=True,
)

ttnn.close_device(device)
```

**Comment by bbradelTT** [comment]
We'll need to pad internally to use the full size of dst, like what we did in layer norm.
